Linux Compile Issue

### Priority

Medium

### Area

- [ ] Data
- [X] Source
- [ ] Docker
- [X] Other

### What happened?

Was following the Compiling on Arch Linux guide. Current issue is 
```
-- Enabled: ipo
-- Found Protobuf Compiler: /usr/bin/protoc
FATAL_ERROR,Failed to generate protobuf files
-- Bot protection: ON
-- Use precompiled header: ON
-- Enabled: Build unity for speed up compilation
-- Disabled: asan
-- Disabled: DEBUG LOG
-- Build type: 
-- Build commit: 
-- Build revision: 
CMake Deprecation Warning at /usr/lib/cmake/OpenAL/OpenALConfig.cmake:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.
Call Stack (most recent call first):
  src/CMakeLists.txt:217 (find_package)


CMake Error at src/CMakeLists.txt:219 (find_package):
  Could not find a package configuration file provided by "Vorbis" with any
  of the following names:

    VorbisConfig.cmake
    vorbis-config.cmake

  Add the installation prefix of "Vorbis" to CMAKE_PREFIX_PATH or set
  "Vorbis_DIR" to a directory containing one of the above files.  If "Vorbis"
  provides a separate development package or SDK, be sure it has been
  installed.


-- Configuring incomplete, errors occurred!
```
when trying to run cmake . -B build

I installed all the required dependency packages but ran into a couple missing dependencies.
Ran

- pacman -S asio
- yay -S stduuid
- yay -S cpp-httplib

to fix the previous errors where it couldn't find those packages, but libvorbis is in the guide already, and tried reinstalling it as well. I know just enough about Arch Linux to get in trouble, and I don't know why this isn't working.

Also the src/framework/protobuf/proto/ doesn't seem to exist in the current repo, but I found src/protobuf/ with a generate.sh in it, and just ran that. Not sure if that is why it is also getting the Fatal_ERROR, Failed to generate protobuf files.

Please let me know if I need to provide more information.

### What OS are you seeing the problem on?

Linux

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -216,7 +216,7 @@ endif()
 if(TOGGLE_FRAMEWORK_SOUND)
 	find_package(OpenAL CONFIG REQUIRED)
 	find_package(VorbisFile REQUIRED)
-	find_package(Vorbis CONFIG REQUIRED)
+	find_package(Vorbis REQUIRED)
 	find_package(Ogg CONFIG REQUIRED)
 endif()
 if(ANDROID)
@@ -471,9 +471,9 @@ if(MSVC)
 		${OPENGL_LIBRARIES}
 		${DirectX_LIBRARY}
 		${DirectX_LIBRARIES}
+		${OGG_LIBRARY}
 		${VORBISFILE_LIBRARY}
 		${VORBIS_LIBRARY}
-		${OGG_LIBRARY}
 		${GMP_LIBRARY}
 		${OPENSSL_LIBRARIES}
 		${DBGHELP_LIBRARY}
@@ -512,6 +512,8 @@ elseif(ANDROID)
 		${NLOHMANN_JSON_LIBRARY}
 		${EGL_LIBRARY}
 		${OGG_LIBRARY}
+		${VORBISFILE_LIBRARY}
+		${VORBIS_LIBRARY}
 		${GMP_LIBRARY}
 		${OpenAL_LIBRARY}
 		${OPENSSL_LIBRARIES}
@@ -521,8 +523,6 @@ elseif(ANDROID)
 		Threads::Threads
 		asio::asio
 		OpenSLES
-		Vorbis::vorbis
-		Vorbis::vorbisfile
 		LibLZMA::LibLZMA
 		game-activity::game-activity
 		GLESv3
@@ -556,6 +556,8 @@ else() # Linux
 		${DirectX_LIBRARY}
 		${DirectX_LIBRARIES}
 		${OGG_LIBRARY}
+		${VORBISFILE_LIBRARY}
+		${VORBIS_LIBRARY}
 		${GMP_LIBRARY}
 		${STDUUID}
 		${FOUNDATION}
@@ -568,8 +570,6 @@ else() # Linux
 		X11::X11
 		asio::asio
 		OpenAL::OpenAL
-		Vorbis::vorbis
-		Vorbis::vorbisfile
 		LibLZMA::LibLZMA
 		pugixml::pugixml
 		ZLIB::ZLIB

